mis_builder/models/mis_builder.py

```diff
@@ -0,0 +1,829 @@
+            move_id = move_obj.search(
+                cr, uid, [('period_id.special', '=', True),
+                          ('period_id.date_start', '<=',
+                           period_to.date_start)],
+                order="period_id desc", limit=1, context=context)
+            if move_id:
+                computed_period_from = move_obj.browse(
+                    cr, uid, move_id[0], context=context).period_id.id
+            else:
+                computed_period_from = self.pool['account.period'].search(
+                    cr, uid, [('company_id', '=',
+                               period_report.company_id.id)],
+                    order='date_start', limit=1)[0]
+            compute_period_ids = period_obj.build_ctx_periods(
+                cr, uid, computed_period_from, period_to.id)
+            domain_list.extend([('period_id', 'in', compute_period_ids)])
+        return domain_list
+
+    def _fetch_queries(self, cr, uid, c, context):
+        res = {}
+        report = c.report_instance_id.report_id
+        for query in report.query_ids:
+            obj = self.pool[query.model_id.model]
+            domain = query.domain and safe_eval(query.domain) or []
+            if query.date_field.ttype == 'date':
+                domain.extend([(query.date_field.name, '>=', c.date_from),
+                               (query.date_field.name, '<=', c.date_to)])
+            else:
+                datetime_from = _utc_midnight(
+                    c.date_from, context.get('tz', 'UTC'))
+                datetime_to = _utc_midnight(
+                    c.date_to, context.get('tz', 'UTC'), add_day=1)
+                domain.extend([(query.date_field.name, '>=', datetime_from),
+                               (query.date_field.name, '<', datetime_to)])
+            if obj._columns.get('company_id', False):
+                domain.extend(['|', ('company_id', '=', False),
+                               ('company_id', '=', c.company_id.id)])
+            field_names = [field.name for field in query.field_ids]
+            obj_ids = obj.search(cr, uid, domain, context=context)
+            obj_datas = obj.read(
+                cr, uid, obj_ids, field_names, context=context)
+            res[query.name] = [AutoStruct(**d) for d in obj_datas]
+        return res
+
+    def _compute(self, cr, uid, lang_id, c, aep, context=None):
+        if context is None:
+            context = {}
+
+        kpi_obj = self.pool['mis.report.kpi']
+
+        res = {}
+
+        localdict = {
+            'registry': self.pool,
+            'sum': sum,
+            'min': min,
+            'max': max,
+            'len': len,
+            'avg': lambda l: sum(l) / float(len(l)),
+        }
+        domain_p = self.compute_period_domain(cr, uid, c, False, False,
+                                              context=context)
+        domain_s = self.compute_period_domain(cr, uid, c, True, False,
```

Comment:
not consistent with defined mode (s instead of e).

What happen if a new mode is added ? Is it possible to get something more generic ?